lazyflow/operators/opCompressedUserLabelArray.py

```diff
@@ -247,12 +262,24 @@ def _executeProjection2D(self, roi, destination):
             # Compute slicing within the deep array and slicing within this block
             deep_relative_intersection = numpy.subtract(intersecting_roi, input_roi.start)
             block_relative_intersection = numpy.subtract(intersecting_roi, block_start)
-                        
-            deep_data = self._getBlockDataset( entire_block_roi )[roiToSlice(*block_relative_intersection)]
+            block_relative_intersection_slicing = roiToSlice(*block_relative_intersection)
+
+            block = self._getBlockDataset( entire_block_roi )
+            deep_data = None
+            if self.Output.meta.has_mask:
+                deep_data = numpy.ma.masked_array(
+                    block["data"][block_relative_intersection_slicing],
+                    mask=block["mask"][block_relative_intersection_slicing],
+                    fill_value=block["fill_value"][()],
+                    shrink=False
+                )
+            else:
+                deep_data = block[block_relative_intersection_slicing]
+
+            # make binary and convert to float (must copy)
+            deep_data_float = deep_data.astype(numpy.float32)
+            deep_data_float[deep_data_float.nonzero()] = 1
 
-            # make binary and convert to float
-            deep_data_float = numpy.where( deep_data, numpy.float32(1.0), numpy.float32(0.0) )
-            
             # multiply by slice-index
```

Comment:
> I used data of shape (100,1000,1000), with deep_data.dtype == uint8. But in this particular case (sparse data), your change is better.

Cool. Just to verify, were you copying the data back into the original array ( [L291](https://github.com/ilastik/lazyflow/pull/164/files#diff-f2aeda246096a7282ca7ee818a962302L291) ) or were you replacing the original ( [L254](https://github.com/ilastik/lazyflow/pull/164/files#diff-f2aeda246096a7282ca7ee818a962302L254) )?